api/src/main/java/org/pmiops/workbench/api/CohortsController.java

```diff
@@ -186,12 +197,51 @@ public Cohort apply(org.pmiops.workbench.db.model.Cohort cohort) {
     // 3. Otherwise, query BigQuery for participant IDs using SQL constructed from the cohort
     // criteria, subject to pagination; remove IDs for participants that had a cohort status not
     // included in the status filter.
-    return ResponseEntity.ok(new MaterializeCohortResponse());
+
+    Workspace workspace = workspaceService.getRequired(workspaceNamespace, workspaceId);
+    CdrVersion cdrVersion = workspace.getCdrVersion();
+    if (request.getCdrVersionName() != null) {
+      cdrVersion = cdrVersionDao.findByName(request.getCdrVersionName());
+      if (cdrVersion == null) {
+        throw new BadRequestException(String.format("Couldn't find CDR version with name %s",
+            request.getCdrVersionName()));
+      }
+    }
+    String cohortSpec;
+    if (request.getCohortName() != null) {
+      org.pmiops.workbench.db.model.Cohort cohort =
+          cohortDao.findCohortByNameAndWorkspaceId(request.getCohortName(), workspace.getWorkspaceId());
+      if (cohort == null) {
+        throw new BadRequestException(
+            String.format("Couldn't find cohort with name %s in workspace %s/%s",
+                request.getCohortName(), workspaceNamespace, workspaceId));
+      }
+      cohortSpec = cohort.getCriteria();
```

Comment:
Is a null check valid here? Or will an exception happen in the cohortMaterializationService?

Reply:
This column is not nullable, so no need for a check I think.